pkg/volume/csi/csi_attacher_test.go

```diff
@@ -221,6 +226,7 @@ func TestAttacherAttach(t *testing.T) {
 				status.AttachError = &storage.VolumeError{
 					Message: "attacher error",
 				}
+				fakeWatcher.Error(testType("error"))
```

Comment:
This looks better to me than changing api machinery.

I would also do something like: https://github.com/kubernetes/kubernetes/blob/312eb890e6cf81fab43d88ae569802689a683065/staging/src/k8s.io/apiserver/pkg/endpoints/watch_test.go#L670-L671
so you don't have to create a fake object type

Reply:
Besides that, I'm not sure about the volumeAttachment specific changes, or if this needs to be added to every test case like this, since I'm not familiar with either of these pieces of code